rotkehlchen/chain/ethereum/names.py

```diff
@@ -67,56 +70,171 @@ def search_for_addresses_names(
         addresses: List[ChecksumEvmAddress],
 ) -> Dict[ChecksumEvmAddress, str]:
     """This method searches for all names of provided addresses known to rotki. We can show
-    only one name per address, and thus we prioritize known names. Priority is the following:
-    blockchain account labels -> private addressbook -> global addressbook
-    -> ethereum tokens -> hardcoded mappings -> ENS names.
+    only one name per address, and thus we prioritize known names. Priority is read from settings.
     """
-    db_addressbook = DBAddressbook(database)
-    db_ens = DBEns(database)
+
+    prioritizer = NamePrioritizer()
+    prioritizer.add_fetchers({
+        'blockchain_account': BlockchainAccountLabelFetcher(database),
+        'global_addressbook': GlobalAddressBookFetcher(database),
+        'private_addressbook': PrivateAddressBookFetcher(database),
+        'ethereum_tokens': TokenMappingsFetcher(database),
+        'hardcoded_mappings': HardcodedAddressFetcher(database),
+        'ens_names': ENSFetcher(database),
+    })
 
     with database.conn.read_ctx() as cursor:
-        reverse_ens = db_ens.get_reverse_ens(cursor, addresses=addresses).values()
-        mapped_tokens = GlobalDBHandler().get_tokens_mappings(addresses=addresses)
-        hardcoded_mappings = {}
-        for address in addresses:
-            constant_name = ETHADDRESS_TO_KNOWN_NAME.get(address)
-            if constant_name is not None:
-                hardcoded_mappings[address] = constant_name
+        settings = database.get_settings(cursor)
+        address_name_priority = settings.address_name_priority
 
-        with db_addressbook.read_ctx(AddressbookType.GLOBAL) as global_cursor:
-            global_addressbook = db_addressbook.get_addressbook_entries(
-                cursor=global_cursor,
-                addresses=addresses,
-            )
-        private_addressbook = db_addressbook.get_addressbook_entries(
-            cursor=cursor,
-            addresses=addresses,
-        )
+    prioritized_addresses = prioritizer.get_prioritized_names(address_name_priority, addresses)
+
+    return prioritized_addresses
+
+
+class NameFetcher(ABC):
+    """
+    Each source, where a name of an address might be stored, is represented by an
+    implementation of this abstract class.
+    """
+    _db: DBHandler
+
+    def __init__(self, db: DBHandler):
+        self._db = db
+
+    @abstractmethod
+    def get_addresses_names(self, addresses: List[ChecksumEvmAddress]) \
+            -> Dict[ChecksumEvmAddress, str]:
+        pass
+
+
+class NamePrioritizer:
```

Comment:
1. Style: For the style issue, is there a definition on when to split imports on multiple lines? I now run the `isort` with the parameters `--wrap-length=99 --line-length=99 --multi-line=3` to have it the way it is styled in other places but it won't catch the 99 columns long import in the `test_ethereum_names.py`.
2. You're right. The ABC is a bit over engineered in this place. I thought the global/private addressbook is also used for BTC and other chains but foud out today that its only used for ETH. My primary intention was to separate the name finding logic from the database access stuff for better testability/SRP.
3. Iterating over the adresses and query every source for that specific address was indeed my first approach. But I took a look at the previous implementation where each data source is queried with all addresses and I thought it is done that way because of database performance issues when calling every source for every address, so to say `M x N` data accesses instead of two for loops.

So I'll change it to the function based / the one address db query approach. Unfortunately, I'm busy with other stuff the next days, so it'll take some time to change that. Just in case you're wondering :)

Reply:
> 1. Style: For the style issue, is there a definition on when to split imports on multiple lines? I now run the `isort` with the parameters `--wrap-length=99 --line-length=99 --multi-line=3` to have it the way it is styled in other places but it won't catch the 99 columns long import in the `test_ethereum_names.py`.

For imports isort is fine. It's one of the few things we have completely automated.


> 3. Iterating over the adresses and query every source for that specific address was indeed my first approach. But I took a look at the previous implementation where each data source is queried with all addresses and I thought it is done that way because of database performance issues when calling every source for every address, so to say `M x N` data accesses instead of two for loops.

No there is no such concern.